src/Parser/MarksmanshipHunter/Modules/Spells/Trueshot.js

```diff
@@ -0,0 +1,40 @@
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS';
+
+class QuickShot extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  trueShotCDReduction = 0;
+  totalFromCD = 0;
+
+  get tsAvailableCasts() {
+    const player = this.combatants.selected;
+    const quickShotRank = player.traitsBySpellId[SPELLS.QUICK_SHOT_TRAIT.id];
+
+    if (quickShotRank === 0) {
+      return;
+    }
+    else if (quickShotRank > 0 && quickShotRank < 4) {
+      this.trueShotCDReduction = quickShotRank * 10;
+    } else if (quickShotRank === 4) {
+      this.trueShotCDReduction = 38;
+    }
+    else if (quickShotRank === 5) {
+      this.trueShotCDReduction = 45;
+    }
+    else if (quickShotRank === 6) {
+      this.trueShotCDReduction = 52;
+    }
+    else if (quickShotRank === 7) {
+      this.trueShotCDReduction = 58;
+    }
+    const cooldownWithTraits = 180 - this.trueShotCDReduction;
+    const tsAvailableCasts = (this.owner.fightDuration / 1000) / cooldownWithTraits;
+    return tsAvailableCasts;
```

Comment:
Now I'm not sure if it wouldn't mess up with the computation in CastEfficiency, but this method should probably get the amount of available casts as an integer, not return something like "1.43627Z6 available casts". Might want to round up or down (`Math.ceil() or Math.floor()`) before returning.

Reply:
Imo best to do all rounding when you display, and to leave values unfounded during calculations so you don't lose any precision you don't need to lose

Reply:
That's a fair point and I'd do that usually too although in this case ("get available casts") it does make sense to round the result I think. I don't know, maybe it doesn't really make a difference.